uni/lib/json.icn

```diff
@@ -991,3 +998,233 @@ end
 ########################
 # END PARSER FUNCTIONS #
 ########################
+
+link fullimag
+
+
+#<p>
+# validate a json file using a json sechema
+# <[param jfname - the name of the json File]>
+# <[param sfname - the name of the json Schema File]>
+#</p>
+#
+procedure jsf_validate(jfname, sfname, strict, error)
+   local json_date, json_schema
+
+   jerror := ErrorHandler(error,strict)
+
+   json_data   := jtou(jfname , "f", strict, error)
+   json_schema := jtou(sfname , "f", strict, error)
+
+   return _jsvalidation(json_data, json_schema, json_schema)
+end
+
+procedure jss_validate(data, schema, strict, error)
+   local json_date, json_schema
+
+   jerror := ErrorHandler(error,strict)
+
+   json_data   := jtou(data , "s", strict, error)
+   json_schema := jtou(schema , "s", strict, error)
+
+   return _jsvalidation(json_data, json_schema, json_schema)
+end
+
+#<p>
+# 
+# Vaildate a specific value only at the curent level given a schema of that value.
+#
+# <[param x is the table]>
+# <[param val is the name of the value of a property]>
+# 
+# <[returns "" if the valid otherwise returs the error string]>
+#
+#</p>
+
+procedure validate_property(x, val)
+   z := ""
+   case x["type"] of {
+      "object" :{
+         if type(val) ~== "table" then {
+            return "expected object type, got: " || val
+            }
+         req := x["required"]
+         if \req then {
+            i := 1
+            cr := 0
+            while i <= *req do {
+               c := 0
+               every prop := key(val) do{
+                  if prop ~== req[i] then c := c + 1
```

Comment:
Is there a reason for not using augmented operators? (this comment applies to several places in this procedure)